### DISCUSSION TITLE
MacOS: Exception and so app crash in libsk4d.dylib

### DISCUSSION THREAD

**Discussion by philippjhofmann** [skia4delphi/skia4delphi#230] [Q&A]
Hi,
I've migrated our app to SKIA but on MacOS sometimes I get the following exception and so app crash in 
skia::textlayout::TextWrapper::breakTextIntoLines:

2023-04-26 10:21:31.007 icTrainer[1675:13183] 10:21:31.007 155902592-Error Exception from TgoExceptionReporter (showInfo:false): 
Zugriffsverletzung bei Adresse 000000011B28A36B beim Zugriff auf Adresse 00000000000002D0
At address: $000000011B28A36B (skia::textlayout::Cluster::runOrNull() const + 11)

Call stack:
icTrainer                 $0000000103DA57C8 Myerrorreporting.TgoExceptionReporter.GlobalGetExceptionStackInfo(TExceptionRecord*) + 136
icTrainer                 $0000000102929E11 Sysutils.Exception.RaisingException(TExceptionRecord*) + 65
icTrainer                 $000000010295A85B Sysutils.RaiseExceptObject(TExceptionRecord*) + 59
icTrainer                 $000000010290A825 _RaiseAtExcept(TObject*, Pointer) + 69
icTrainer                 $0000000102928C51 SignalConverter + 49
icTrainer                 $0000000102928C20 SignalConverter + 0
libsk4d.dylib             $000000011B2949C5 skia::textlayout::TextWrapper::breakTextIntoLines(skia::textlayout::ParagraphImpl*, float, std::__1::function<void (skia::textlayout::SkRange<unsigned long>, skia::textlayout::SkRange<unsigned long>, skia::textlayout::SkRange<unsigned long>, skia::textlayout::SkRange<unsigned long>, skia::textlayout::SkRange<unsigned long>, float, unsigned long, unsigned long, SkPoint, SkPoint, skia::textlayout::InternalLineMetrics, bool)> const&) + 341
libsk4d.dylib             $000000011B284419 skia::textlayout::ParagraphImpl::breakShapedTextIntoLines(float) + 393
libsk4d.dylib             $000000011B28368B skia::textlayout::ParagraphImpl::layout(float) + 699
icTrainer                 $000000010421EC07 Skia.TSkParagraph.Layout(Single) + 39
icTrainer                 $00000001042376BD Skia.Fmx.Graphics.TSkTextLayout.UpdateParagraph().ParagraphLayout(Pointer, Skia.ISkParagraph, Single) + 237
icTrainer                 $0000000104237957 Skia.Fmx.Graphics.TSkTextLayout.UpdateParagraph().DoUpdateParagraph(Pointer, var Skia.Fmx.Graphics.TSkTextLayout.TParagraph, String, Integer) + 263
icTrainer                 $000000010423390C Skia.Fmx.Graphics.TSkTextLayout.UpdateParagraph() + 1180
icTrainer                 $000000010422D9D0 Skia.Fmx.Graphics.TSkTextLayout.DoRenderLayout() + 48
icTrainer                 $0000000102C13D4C Fmx.Textlayout.TTextLayout.EndUpdate() + 76
icTrainer                 $0000000102D6D12D Fmx.Objects.TText.Paint() + 2221
icTrainer                 $0000000102DF56D0 Fmx.Controls.TControl.PaintInternal().DoPaintInternal(Pointer) + 960
icTrainer                 $0000000102DF5E5C Fmx.Controls.TControl.PaintInternal().PaintAndClipChild(Pointer) + 1068
icTrainer                 $0000000102DF43C9 Fmx.Controls.TControl.PaintInternal() + 153
icTrainer                 $0000000102DD75A9 Fmx.Controls.TControl.PaintChildren() + 1033
Runtime error 217 at 000000011B28A36B
2023-04-26 10:21:31.014 icTrainer[1675:13183] 10:21:31.014 155902592-Info FormDestroy
Runtime error 217 at 0000000103274047
objc[1675]: objc_disposeClassPair: class 'FMXWindow' still has subclasses, including 'NSKVONotifying_FMXWindow'!
objc[1675]: objc_disposeClassPair: class 'FMXWindow' still has subclasses, including 'NSKVONotifying_FMXWindow'!
Runtime error 217 at 00000001028F9875

How this could be handeled? Best regards, Philipp

**Comment by viniciusfbb** [comment]
Fixed on last version **[v6-beta](https://github.com/skia4delphi/skia4delphi/releases)**
